web/modules/custom/collection/src/ContextProvider/CollectionRouteContext.php

```diff
@@ -88,20 +87,30 @@ public function getRuntimeContexts(array $unqualified_context_ids) {
     $cacheability = new CacheableMetadata();
     $cacheability->setCacheContexts(['route']);
 
-    $context = new Context($context_definition, $value);
+    $collection_context_definition = new ContextDefinition('entity:rdf_entity:collection', NULL, FALSE);
+    $context = new Context($collection_context_definition, $value);
     $context->addCacheableDependency($cacheability);
-
     $result['collection'] = $context;
 
+    // Also create a context definition to be available for the og_menu block.
+    $collection_context_definition = new ContextDefinition('entity', $this->t('Organic group provided by collection'), FALSE);
+    $context = new Context($collection_context_definition, $value);
+    $context->addCacheableDependency($cacheability);
+    $result['og'] = $context;
```

Comment:
Is it better? If you consider that we do not load anything two times (because we are assigning one value in two indexes) there is not much impact.
On the other hand, the og context is to be able to use the api of a contrib module (i.e. og_menu) because it's context does not cover our needs in total. On our side, we have blocks that specifically need the collection context. If we leave both, we keep these functionalities separate.
I can remove the collection context though and only provide the og.
We cannot use the context from the og_menu as it does not extend the context to the group entities.